silkworm/infra/common/timer_test.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+   Copyright 2022 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "timer.hpp"
+
+#include <string>
+#include <thread>
+
+#include <boost/asio/io_context.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+namespace silkworm {
+
+TEST_CASE("Timer lifecycle race condition") {
+    struct TimerInfo {
+        std::string description;
+        uint32_t interval;
+    };
+    const std::vector<TimerInfo> timer_infos{
+        {"long_running_timer", 10'000},
```

Comment:
Is this gonna wait for 10 sec? If yes, I'd like to avoid this. Perhaps removing this line is good enough?

Reply:
> Is this gonna wait for 10 sec? If yes, I'd like to avoid this. Perhaps removing this line is good enough?

It wasn't waiting for 10 sec because it didn't expired. In any case, I've improved the unit tests and also reduced the timer durations, so it's not an issue anymore.